cloud-ml/c3d/trainer/task.py

```diff
@@ -0,0 +1,142 @@
+"""
+This script serves serves as a way to train the C3D model on subsets of the
+training data of varying sizes, reporting the results of that training to
+the model_results channel on Slack.
+"""
+import datetime
+import logging
+import subprocess
+
+import keras
+import numpy as np
+
+from blueno import gcs as plotting
+from blueno import utils
+from .model import C3DBuilder
+
+
+def train_model(x_train: np.ndarray, y_train: np.ndarray, x_valid: np.ndarray,
+                y_valid: np.ndarray):
+    """
+    Train a C3D model on the given data.
+
+    :param x_train:
+    :param y_train:
+    :param x_valid:
+    :param y_valid:
+    :return:
+    """
+    # To avoid data type issues in the Keras library calls
+    if x_train.shape[1:] != (32, 32, 32) or x_valid.shape[1:] != (32, 32, 32):
+        raise ValueError(
+            'x_train and x_valid should have shape (?, 32, 32,32), got {} '
+            'and {}'.format(
+                x_train.shape, x_valid.shape
+            ))
+
+    if y_train.ndim != 1 or y_valid.ndim != 1:
+        raise ValueError(
+            'y_train and y_valid should have shape (?,), got {} and {}'.format(
+                y_train.shape, y_valid.shape
+            ))
+
+    x_train = x_train.astype(np.float32)
+    y_train = y_train.astype(np.int16)
+    x_valid = x_valid.astype(np.float32)
+    y_valid = y_valid.astype(np.int16)
+
+    x_train = np.expand_dims(x_train, axis=-1)
+    x_valid = np.expand_dims(x_valid, axis=-1)
+    y_train = np.expand_dims(y_train, axis=-1)
+    y_valid = np.expand_dims(y_valid, axis=-1)
+
+    mean = x_train.mean()
+    std = x_train.std()
+    x_train = (x_train - mean) / std
+    x_valid = (x_valid - mean) / std
+
+    describe_data(x_train, y_train, x_valid, y_valid)
+
+    metrics = ['acc',
+               utils.true_positives,
+               utils.false_negatives,
+               utils.sensitivity,
+               utils.specificity]
+    model = C3DBuilder.build()
+    opt = keras.optimizers.Adam(lr=1e-4)
```

Comment:
Not sure how this will train — the successful C3D that I put together used SGD with LR = 1e-5.